core/src/main/scala/org/ensime/core/Refactoring.scala

```diff
@@ -24,7 +25,16 @@ abstract class RefactoringEnvironment(file: String, start: Int, end: Int) {
     procId: Int,
     tpe: RefactorType,
     parameters: refactoring.RefactoringParameters
-  ): Either[RefactorFailure, RefactorEffect] = {
+  )(implicit charset: Charset): Either[RefactorFailure, RefactorDiffEffect] = {
```

Comment:
do we really need to pass around the `Charset` everywhere? Are we not free to choose the encoding of the diff file?